### ISSUE TITLE
<br> and <p>

### ISSUE THREAD

**Issue by jnmwarren** [RickStrahl/MarkdownMonster#484]
I upgraded from 1.14.12 to 1.15.00 and I noticed now when I press the `<Enter>` key I get a `<br>` and not a `<p>` as the previous version. Is there a way to change it back to were I get the `<p>` when the `<Enter>` key is pressed and `<br>` when `<Shift>+<Enter>` is pressed.

**Comment by RickStrahl** [comment]
Ok not sure what characters you mean as they are not making it into the message here.

**Comment by jnmwarren** [comment]
When coping the Markdown as HTML the code gets copied as follows.

1.14.12
<p>OpenDate: 20190122</p>
<p>ClosedDate: 20190125</p>

1.15.00
<p>OpenDate: 20190201<br />
<br>ClosedDate: 20190205</p>

How do you prevent the <br> tag and only use the <p> tag?

**Comment by jnmwarren** [comment]
When coping Markdown to HTML code you get the following.

1.14.12
> <p>OpenDate: 20190122</p>
> <p>ClosedDate: 20190125</p>

1.15.00
> <p>OpenDate: 20190201<br />
> <br>ClosedDate: 20190205</p>

How do you prevent the <br> tag and only use the <p> tag?

**Comment by jnmwarren** [comment]
When coping Markdown to HTML code you get the following.

1.14.12
```<p>OpenDate: 20190122</p>
```<p>ClosedDate: 20190125</p>

1.15.00
```<br>OpenDate: 20190201<br />
```<br>ClosedDate: 20190205</p>

```How do you prevent the <br> tag and only use the <p> tag?

**Comment by jnmwarren** [comment]
When I copied the Markdown as HTML on get the following:
1.14.12
```<p>Closed Date</p>

1.15.00
```<p>Closed Date<br />
```<br>Open Date</p>
'''How do you prevent the use of the <br> tags and only use the <p> tags?

**Comment by RickStrahl** [comment]
I still don't understand what you're asking. You're copying what and you're getting what?

**Comment by RickStrahl** [comment]
Ok I looked at your raw questions and fixed the markup so we can understand what you're talking about.

The problem is you're not using valid Markdown or not using valid Markdown formatting for what you're wanting to do.

Single line breaks in Markdown text **are not broken at all** - they are treated as spaces and are part of the same paragraph. To get a `<p>` you need two line breaks. That hasn't changed and that has really nothing to do with MM - that's just the way Markdown works.

So to summarize:

* `<p>`  -  to line breaks
* `<br>` -  `\` or `  ` (two spaces) at the end of the line
* no break at all - plain text or single line break

Still not sure what your sample is trying to demonstrate - whatever that is it looks wrong :-) the triple backticks are used for code blocks so your code should be **inside** of those tags, starting on the next line and ending before the last set of triple ticks.

**Comment by jnmwarren** [comment]
You have answered my question. Thanks for the code block explanation. I was trying to give an example as best as I could.

**Comment by jnmwarren** [comment]
One other question along the same lines. Why would MM put in a <br> instead of <p> when the Enter key is used?

**Comment by jnmwarren** [comment]
One other question. Why in MM when I press the Enter key would I get a <code>&lt;br&gt;</code> and not a <code>&lt;p&gt;</code>? Shouldn't MM render a Shift + Enter as a <code>&lt;br&gt;</code> and not an Enter

**Comment by RickStrahl** [comment]
I think you're asking the same question you asked previously - and the answer is still the same. Please take another look at the answer above.

**Comment by jnmwarren** [comment]
Thanks

**Comment by RickStrahl** [comment]
Just for clarification, maybe the following will help:

![image](https://user-images.githubusercontent.com/1374013/52530528-770ed000-2cab-11e9-98ac-c5d895c9f058.png)

I've turned on **View Invisible Characters** so you can see the spaces and returns etc.

I've also added a hotkey for `Shift-Enter` that enters the key-combo for a soft return (ie. a single linebreak) in the latest preview release.

**Comment by jnmwarren** [comment]
Yes, that does help.
